src/common/replication_common.cpp

```diff
@@ -569,6 +498,118 @@ void replica_helper::load_meta_servers(/*out*/ std::vector<dsn::rpc_address> &se
     dassert(servers.size() > 0, "no meta server specified in config [%s].%s", section, key);
 }
 
+/*static*/ bool
+replication_options::get_data_dir_and_tag(const std::string &config_dirs_str,
+                                          const std::string &default_dir,
+                                          const std::string &app_name,
+                                          /*out*/ std::vector<std::string> &data_dirs,
+                                          /*out*/ std::vector<std::string> &data_dir_tags,
+                                          /*out*/ std::string &err_msg)
+{
+    // - if {config_dirs_str} is empty (return true):
```

Comment:
I think maybe it will be better to make it a `void` function, if some configurations are invalid, just `assert` and return. That will be easier to understand and no need for complicated comments and conditional judgments.

Reply:
Yes, I agree that make it a `void` function is simple. I used to implement it as a `void` function, but I found that it was hard to write a unit test for this function. Finally, I update my implmentation.

Reply:
Maybe the gtest macro `ASSERT_NO_FATAL_FAILURE` is useful for assertion tests.

Reply:
I tried to use `ASSERT_NO_FATAL_FAILURE`, but it seems not work and still core when `dassert`. I have read the usage of `ASSERT_NO_FATAL_FAILURE`
```
// Macros that execute statement and check that it doesn't generate new fatal failures in the current thread
```
It seems that using `ASSERT_NO_FATAL_FAILURE` can not avoid assert in source code.